### PR TITLE
Add comment youyan

### PR DESCRIPTION
**Pull request by jinfang134**
for issue #1305

add youyan comments for next themes
添加友言评论支持，

for more detail,please read my blog.
更多细节请访问我的博客http://jinfang.oschina.io/posts/5f6c0475/

### CODE REVIEW

File: _config.yml [hunk 1]
```diff
@@ -277,6 +277,9 @@ mathjax:
 # Gentie productKey
 #gentie_productKey:
 
+# youyan uid
```
Comment (ivan-nginx) [inline _config.yml]: Link to get youyan uid in comment maybe?

Comment (jinfang134) [inline _config.yml]: login uyan.cc
then get your code 
e.g.
```
<!-- UY BEGIN -->
<div id="uyan_frame"></div>
<script type="text/javascript" src="http://v2.uyan.cc/code/uyan.js?uid=34343324"></script>
<!-- UY END -->
```
then just fill your uid get from uyan.cc in the _config.yml
e.g.
youyan_uid=34343324

Comment (jinfang134) [inline _config.yml]: the url of youyan is http://uyan.cc
and it is just in Chinese

Comment (ivan-nginx) [inline _config.yml]: Thanks.
In `_config.yml` this comment in short variant maybe?

Comment (jinfang134) [inline _config.yml]: I write a blog to explain how to implement ,
https://jinfang134.github.io/posts/5f6c0475/
sorry for it is writen in chinese.

Comment (jinfang134) [inline _config.yml]: just fill in the `youyan_uid` in the `_config.yml` for adding youyan comment in your hexo

File: layout/_scripts/third-party/comments/youyan.swig [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+  {% endif %}
+
+  {% if page.comments %}
+      <!-- UY BEGIN -->
+     
+      <script type="text/javascript" src="http://v2.uyan.cc/code/uyan.js?uid={{uid}}"></script>
+      <!-- UY END -->    
+  {% endif %}
+
+{% endif %}
```
Comment (geekrainy) [inline layout/_scripts/third-party/comments/youyan.swig]: You can add a new line at end of file.

File: layout/_partials/comments.swig [hunk 3]
```diff
@@ -22,6 +22,8 @@
       <div id="hypercomments_widget"></div>
     {% elseif theme.gentie_productKey %}
       <div id="cloud-tie-wrapper" class="cloud-tie-wrapper"></div>
+    {% elseif theme.youyan_uid %}
+         <div id="uyan_frame"></div>
```
Comment (geekrainy) [inline layout/_partials/comments.swig]: Keep indent 2 spaces.  :)

File: _config.yml [hunk 4]
```diff
@@ -277,8 +277,11 @@ mathjax:
 # Gentie productKey
 #gentie_productKey:
 
-# youyan uid
-#youyan_uid: 
+# support youyan comments,
+# you can regist and get your uid from http://www.uyan.cc
+# for more informations, you can visit my blog https://jinfang134.github.io/posts/5f6c0475/
```
Comment (ivan-nginx) [inline _config.yml]: No need to PR your blog in config file. All info about usage must be in conversatioan. For example, u can see my pulls.

Comment (jinfang134) [inline _config.yml]: ok,i modify it

File: layout/_partials/comments.swig (hunk 3 above)
Comment (jinfang134) [inline layout/_partials/comments.swig]: updated

File: layout/_scripts/third-party/comments.swig [hunk 5]
```diff
@@ -2,3 +2,4 @@
 {% include './comments/disqus.swig' %}
 {% include './comments/hypercomments.swig' %}
 {% include './comments/gentie.swig' %}
+{% include './comments/youyan.swig' %}
```
Comment (geekrainy) [inline layout/_scripts/third-party/comments.swig]: An new at end of the file.

File: _config.yml [hunk 6]
```diff
@@ -277,6 +277,11 @@ mathjax:
 # Gentie productKey
 #gentie_productKey:
 
+# support youyan comments,
+# you can regist and get your uid from http://www.uyan.cc
+# youyan uid ,like youyan_id=2343223
```
Comment (ivan-nginx) [inline _config.yml]: -youyan uid ,like youyan_id=2343223
+youyan uid, like youyan_id=2343223

Comment (jinfang134) [inline _config.yml]: ???

Comment (ivan-nginx) [inline _config.yml]: Fix on this:
# Support for youyan comments system.
# You can get your uid from http://www.uyan.cc

Comment (jinfang134) [inline _config.yml]: what is meaning?

Comment (jinfang134) [inline _config.yml]: ok ,i know!

Comment (jinfang134) [inline _config.yml]: haa, my English is so poor...

Comment (jinfang134) [inline _config.yml]: updated!

**Comment by ivan-nginx** [comment]
Well done. For now code looks fine. Now nedd just wait for merge.

And Happy New Year to everybody! :>

**Comment by jinfang134** [comment]
Happy new year.
I have five days holiday.

**Comment by ehlxr** [comment]
请问，如何此加上评论数的显示image

**Comment by jinfang134** [comment]
方法来自next的官方文档

http://theme-next.iissnan.com/third-party-services.html#analytics-system

里的链接


https://notes.wanghao.work/2015-10-21-%E4%B8%BANexT%E4%B8%BB%E9%A2%98%E6%B7%BB%E5%8A%A0%E6%96%87%E7%AB%A0%E9%98%85%E8%AF%BB%E9%87%8F%E7%BB%9F%E8%AE%A1%E5%8A%9F%E8%83%BD.html#%E9%85%8D%E7%BD%AELeanCloud

**Comment by jinfang134** [comment]
至于评论数，应该没有办法添加，因为使用的第三方的评论，不是自建的。

**Comment by ehlxr** [comment]
@jinfang134 多谢，多说也是第三方的，目前可以显示

**Comment by monotone** [comment]
@jinfang134 友言评论功能失效了吗？ monotone.github.io

**Pull request merged**